for similar issues with port gtk-vnc and this solution, see:

https://gitlab.gnome.org/GNOME/gtk-vnc/-/issues/4
and
https://gitlab.gnome.org/GNOME/gtk-vnc/-/commit/a1080abbdcdf5261fb2b9e5523119640f09d1e34

--- src/mmap.hpp.orig	2021-05-09 23:46:35.000000000 -0400
+++ src/mmap.hpp	2021-05-09 23:47:48.000000000 -0400
@@ -53,6 +53,11 @@
 # include <limits>
 #endif

+// Back compat for OSX 10.10 and earlier
+#ifndef MAP_ANONYMOUS
+#define MAP_ANONYMOUS MAP_ANON
+#endif
+
 // manage mmap state
 class MMap {

